🐞  MSYS2/clang64 CI build fails: std::char_traits<unsigned char> not defined in newer libc++

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

The windows-2022-msys2-clang-14 matrix entry in the pull-request.yml workflow fails to compile with errors like:

```cpp
D:/a/electionguard-core2/electionguard-core2/.cache/CPM/nlohmann_json/3cc46c5b3b30780707857320f5d2655b293c4330/include/nlohmann/detail/input/input_adapters.hpp:148:19: error: implicit instantiation of undefined template 'std::char_traits<unsigned char>'
  148 |     typename std::char_traits<char_type>::int_type get_character()
```

The errors originate from nlohmann/json's using std::char_traits<unsigned char> — a specialization that is not required by the C++ standard and is not provided by the newer libc++ shipped with the MSYS2 clang64 toolchain.

The project currently pins nlohmann/json 3.11.2 via CPM in src/CMakeLists.txt. This version still relies on std::char_traits<unsigned char>.

### Expected Behavior

_No response_

### Steps To Reproduce

1. Push to a branch that triggers the pull-request.yml workflow
2. Observe the windows-2022-msys2-clang-14 job failure in the build step (make build-msys2)


### Environment

```markdown
•  CI: GitHub Actions, windows-2022 runner with MSYS2 clang64
•  Compiler: Clang 14 (MSYS2 clang64 toolchain)
```

### Anything else?

_No response_

--- a/.config/dotnet-tools.json
+++ b/.config/dotnet-tools.json
@@ -9,7 +9,7 @@
       ]
     },
     "nvika": {
-      "version": "3.0.0",
+      "version": "4.0.0",
       "commands": [
         "nvika"
       ]

--- a/.config/dotnet-tools.json
+++ b/.config/dotnet-tools.json
@@ -3,7 +3,7 @@
   "isRoot": true,
   "tools": {
     "jetbrains.resharper.globaltools": {
-      "version": "2022.3.2",
+      "version": "2025.3.3",
       "commands": [
         "jb"
       ]

--- a/.github/workflows/pull-request-ui.yml
+++ b/.github/workflows/pull-request-ui.yml
@@ -43,7 +43,7 @@ jobs:
       source: ${{ steps.filter.outputs.source }}
     steps:
       # For pull requests it's not necessary to checkout the code
-      - uses: dorny/paths-filter@v2
+      - uses: dorny/paths-filter@v3
         id: filter
         with:
           filters: |

--- a/.github/workflows/pull-request-ui.yml
+++ b/.github/workflows/pull-request-ui.yml
@@ -69,8 +69,8 @@ jobs:
         name: ["macOS", "windows-os"]
         include:
           - name: macOS
-            os: macos-15
-            version: "16.4"
+            os: macos-26
+            version: "26.2"
             osFolder: "maccatalyst"
           - name: windows-os
             os: windows-2022

--- a/.github/workflows/pull-request-ui.yml
+++ b/.github/workflows/pull-request-ui.yml
@@ -11,7 +11,7 @@ env:
 jobs:
   get_version_code:
     name: Get Version Code
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
     outputs:
       version: ${{ steps.get_version_code.outputs.version }}
       build_number: ${{ steps.get_version_code.outputs.build_number }}

--- a/.github/workflows/pull-request-ui.yml
+++ b/.github/workflows/pull-request-ui.yml
@@ -132,20 +132,19 @@ jobs:
       - name: Clean UI
         if: (steps.filter.outputs.bindings == 'true' || steps.filter.outputs.source == 'true')
         run: make clean-ui
-
-      - name: Publish UI
-        if: (steps.filter.outputs.bindings == 'true' || steps.filter.outputs.source == 'true')
-        env:
-          BUILD_NUMBER: ${{ needs.get_version_code.outputs.build_number }}
-          BUILD_VERSION: "${{needs.get_version_code.outputs.version}}"
-          APPCENTER_SECRET_MACOS: ${{ secrets.APPCENTER_SECRET_MACOS }}
-          APPCENTER_SECRET_UWP: ${{ secrets.APPCENTER_SECRET_UWP }}
-        run: make publish-ui
-
-      - name: save build artifacts
-        if: (steps.filter.outputs.bindings == 'true' || steps.filter.outputs.source == 'true')
-        uses: actions/upload-artifact@v4
-        with:
-          name: electionguard-admin
-          retention-days: 7
-          path: ./publish/*
+#      - name: Publish UI
+#        if: (steps.filter.outputs.bindings == 'true' || steps.filter.outputs.source == 'true')
+#        env:
+#          BUILD_NUMBER: ${{ needs.get_version_code.outputs.build_number }}
+#          BUILD_VERSION: "${{needs.get_version_code.outputs.version}}"
+#          APPCENTER_SECRET_MACOS: ${{ secrets.APPCENTER_SECRET_MACOS }}
+#          APPCENTER_SECRET_UWP: ${{ secrets.APPCENTER_SECRET_UWP }}
+#        run: make publish-ui
+#
+#      - name: save build artifacts
+#        if: (steps.filter.outputs.bindings == 'true' || steps.filter.outputs.source == 'true')
+#        uses: actions/upload-artifact@v4
+#        with:
+#          name: electionguard-admin
+#          retention-days: 7
+#          path: ./publish/*

--- a/.github/workflows/pull-request-ui.yml
+++ b/.github/workflows/pull-request-ui.yml
@@ -69,8 +69,8 @@ jobs:
         name: ["macOS", "windows-os"]
         include:
           - name: macOS
-            os: macOS-13
-            version: "14.2"
+            os: macos-15
+            version: "16.4"
             osFolder: "maccatalyst"
           - name: windows-os
             os: windows-2022
@@ -113,7 +113,7 @@ jobs:
         if: (runner.os == 'macOS' && steps.filter.outputs.any == 'true')
         run: |
           ls -ls /Applications/
-          ls -ls $ANDROID_SDK_ROOT/ndk
+          # ANDROID_SDK_ROOT may not be set on all runners
           sudo xcode-select -switch /Applications/Xcode_${{ matrix.version }}.app
 
       - name: Configure Environment

--- a/.github/workflows/pull-request-ui.yml
+++ b/.github/workflows/pull-request-ui.yml
@@ -73,7 +73,7 @@ jobs:
             version: "26.2"
             osFolder: "maccatalyst"
           - name: windows-os
-            os: windows-2022
+            os: windows-2025-vs2026
             version: "latest"
             osFolder: "windows"
 

--- a/.github/workflows/pull-request.yml
+++ b/.github/workflows/pull-request.yml
@@ -26,7 +26,7 @@ jobs:
       typescript: ${{ steps.filter.outputs.typescript }}
     steps:
       # For pull requests it's not necessary to checkout the code
-      - uses: dorny/paths-filter@v2
+      - uses: dorny/paths-filter@v3
         id: filter
         with:
           filters: |

--- a/.github/workflows/pull-request.yml
+++ b/.github/workflows/pull-request.yml
@@ -70,9 +70,9 @@ jobs:
             "ubuntu-24.04-gcc-12.1.0",
             "macos-15-intel-xcode-16.4",
             "macos-15-xcode-16.4-arm64",
-            "windows-2022-msys2-clang-14",
-            "windows-2022-msvc-latest-x86",
-            "windows-2022-msvc-latest-x64",
+            "windows-2025-msys2-clang-14",
+            "windows-2025-msvc-latest-x86",
+            "windows-2025-msvc-latest-x64",
             "ubuntu-24.04-linter",
           ]
         include:
@@ -121,8 +121,8 @@ jobs:
             lint: false
             runCsTests: false
             runTsTests: false
-          - name: windows-2022-msys2-clang-14
-            os: windows-2022
+          - name: windows-2025-msys2-clang-14
+            os: windows-2025-vs2026
             processor: x64
             compiler: clang
             version: "14"
@@ -130,8 +130,8 @@ jobs:
             lint: false
             runCsTests: false
             runTsTests: false
-          - name: windows-2022-msvc-latest-x86
-            os: windows-2022
+          - name: windows-2025-msvc-latest-x86
+            os: windows-2025-vs2026
             processor: x86
             compiler: msvc
             version: "latest"
@@ -139,8 +139,8 @@ jobs:
             lint: false
             runCsTests: false
             runTsTests: false
-          - name: windows-2022-msvc-latest-x64
-            os: windows-2022
+          - name: windows-2025-msvc-latest-x64
+            os: windows-2025-vs2026
             processor: x64
             compiler: msvc
             version: "latest"

--- a/.github/workflows/pull-request.yml
+++ b/.github/workflows/pull-request.yml
@@ -259,9 +259,9 @@ jobs:
             base-devel
             mingw-w64-clang-x86_64-gcc
             mingw-w64-clang-x86_64-toolchain
-            mingw-w64-clang-x86_64-clang-14
+            mingw-w64-clang-x86_64-clang
             mingw-w64-clang-x86_64-cmake
-            mingw-w64-clang-x86_64-llvm-14
+            mingw-w64-clang-x86_64-llvm
             make
             git
 

--- a/.github/workflows/pull-request.yml
+++ b/.github/workflows/pull-request.yml
@@ -70,7 +70,7 @@ jobs:
             "ubuntu-22.04-gcc-12.1.0",
             "macos-15-intel-xcode-16.4",
             "macos-15-xcode-16.4-arm64",
-            "windows-2022-gcc-11",
+            "windows-2022-msys2-clang-14",
             "windows-2022-msvc-latest-x86",
             "windows-2022-msvc-latest-x64",
             "ubuntu-22.04-linter",
@@ -121,11 +121,11 @@ jobs:
             lint: false
             runCsTests: false
             runTsTests: false
-          - name: windows-2022-gcc-11
+          - name: windows-2022-msys2-clang-14
             os: windows-2022
             processor: x64
-            compiler: gcc
-            version: "11"
+            compiler: clang
+            version: "14"
             makeReleaseBuild: false
             lint: false
             runCsTests: false
@@ -319,6 +319,13 @@ jobs:
           PROCESSOR: ${{ matrix.processor }}
         if: (runner.os == 'Windows' && matrix.compiler == 'msvc' && steps.filter.outputs.cpp == 'true')
         run: make test
+
+      - name: Run Tests (Windows MSYS2)
+        env:
+          PROCESSOR: ${{ matrix.processor }}
+        if: (runner.os == 'Windows' && matrix.compiler == 'clang' && steps.filter.outputs.cpp == 'true')
+        shell: msys2 {0}
+        run: make test-msys2
 
       - name: Run Tests (Windows-x86)
         env:

--- a/.github/workflows/pull-request.yml
+++ b/.github/workflows/pull-request.yml
@@ -65,19 +65,19 @@ jobs:
       matrix:
         name:
           [
-            "ubuntu-22.04-clang-14.0.0",
-            "ubuntu-22.04-emscripten-3.1",
-            "ubuntu-22.04-gcc-12.1.0",
+            "ubuntu-24.04-clang-14.0.0",
+            "ubuntu-24.04-emscripten-3.1",
+            "ubuntu-24.04-gcc-12.1.0",
             "macos-15-intel-xcode-16.4",
             "macos-15-xcode-16.4-arm64",
             "windows-2022-msys2-clang-14",
             "windows-2022-msvc-latest-x86",
             "windows-2022-msvc-latest-x64",
-            "ubuntu-22.04-linter",
+            "ubuntu-24.04-linter",
           ]
         include:
-          - name: ubuntu-22.04-clang-14.0.0
-            os: ubuntu-22.04
+          - name: ubuntu-24.04-clang-14.0.0
+            os: ubuntu-24.04
             processor: x64
             compiler: clang
             version: "14"
@@ -85,8 +85,8 @@ jobs:
             lint: false
             runCsTests: false
             runTsTests: false
-          - name: ubuntu-22.04-emscripten-3.1
-            os: ubuntu-22.04
+          - name: ubuntu-24.04-emscripten-3.1
+            os: ubuntu-24.04
             processor: x64
             compiler: emscripten
             version: "3.1.35"
@@ -94,8 +94,8 @@ jobs:
             lint: false
             runCsTests: false
             runTsTests: true
-          - name: ubuntu-22.04-gcc-12.1.0
-            os: ubuntu-22.04
+          - name: ubuntu-24.04-gcc-12.1.0
+            os: ubuntu-24.04
             processor: x64
             compiler: gcc
             version: "12"
@@ -148,8 +148,8 @@ jobs:
             lint: false
             runCsTests: true
             runTsTests: false
-          - name: ubuntu-22.04-linter
-            os: ubuntu-22.04
+          - name: ubuntu-24.04-linter
+            os: ubuntu-24.04
             processor: x64
             compiler: clang
             version: "14"

--- a/.github/workflows/pull-request.yml
+++ b/.github/workflows/pull-request.yml
@@ -68,8 +68,8 @@ jobs:
             "ubuntu-22.04-clang-14.0.0",
             "ubuntu-22.04-emscripten-3.1",
             "ubuntu-22.04-gcc-12.1.0",
-            "macos-13-xcode-14.2",
-            "macos-13-xcode-14.2-arm64",
+            "macos-15-intel-xcode-16.4",
+            "macos-15-xcode-16.4-arm64",
             "windows-2022-gcc-11",
             "windows-2022-msvc-latest-x86",
             "windows-2022-msvc-latest-x64",
@@ -103,20 +103,20 @@ jobs:
             lint: false
             runCsTests: false
             runTsTests: false
-          - name: macOS-13-xcode-14.2
-            os: macOS-13
+          - name: macos-15-intel-xcode-16.4
+            os: macos-15-intel
             processor: x64
             compiler: xcode
-            version: "14.2"
+            version: "16.4"
             makeReleaseBuild: false
             lint: false
             runCsTests: true
             runTsTests: false
-          - name: macOS-13-xcode-14.2-arm64
-            os: macOS-13
+          - name: macos-15-xcode-16.4-arm64
+            os: macos-15
             processor: arm64
             compiler: xcode
-            version: "14.2"
+            version: "16.4"
             makeReleaseBuild: false
             lint: false
             runCsTests: false
@@ -246,7 +246,7 @@ jobs:
         if: (runner.os == 'macOS' && steps.filter.outputs.any == 'true')
         run: |
           ls -ls /Applications/
-          ls -ls $ANDROID_SDK_ROOT/ndk
+          # ANDROID_SDK_ROOT may not be set on all runners
           sudo xcode-select -switch /Applications/Xcode_${{ matrix.version }}.app
 
       - name: Update Environment (Windows MSYS2)

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -59,9 +59,9 @@ jobs:
 
   maccatalyst_build_arm64:
     name: MacCatalyst Build (arm64)
-    runs-on: macos-13
+    runs-on: macos-15
     env:
-      compiler_version: 14.2
+      compiler_version: 16.4
     steps:
       - name: Checkout Code
         uses: actions/checkout@v4
@@ -82,9 +82,9 @@ jobs:
 
   macos_build_arm64:
     name: MacOS Build (arm64)
-    runs-on: macos-13
+    runs-on: macos-15
     env:
-      compiler_version: 14.2
+      compiler_version: 16.4
     steps:
       - name: Checkout Code
         uses: actions/checkout@v4
@@ -105,9 +105,9 @@ jobs:
 
   macos_build_x64:
     name: MacOS Build (x64)
-    runs-on: macos-13
+    runs-on: macos-15-intel
     env:
-      compiler_version: 14.2
+      compiler_version: 16.4
     steps:
       - name: Checkout Code
         uses: actions/checkout@v4
@@ -313,8 +313,8 @@ jobs:
         name: ["windows-os"]
         include:
           # - name: macOS
-          #   os: macOS-13
-          #   version: "14.2"
+          #   os: macos-15
+          #   version: "16.4"
           #   osFolder: "maccatalyst"
           - name: windows-os
             os: windows-2022

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -11,7 +11,7 @@ env:
 jobs:
   get_version_code:
     name: Get Version Code
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
     outputs:
       version: ${{ steps.get_version_code.outputs.version }}
       build_number: ${{ steps.get_version_code.outputs.build_number }}
@@ -32,7 +32,7 @@ jobs:
 
   linux_build_x64:
     name: Linux Build (x64)
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
     env:
       compiler: clang
       compiler_version: 14
@@ -128,7 +128,7 @@ jobs:
 
   wasm_build:
     name: WASM Build
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
     env:
       compiler: clang
       compiler_version: 14
@@ -216,7 +216,7 @@ jobs:
         windows_build_x64,
         windows_build_x86,
       ]
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
     steps:
       - name: Checkout Code
         uses: actions/checkout@v4
@@ -275,7 +275,7 @@ jobs:
   publish_npm:
     name: Publish NPM
     needs: [get_version_code, wasm_build]
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
     steps:
       - name: Checkout Code
         uses: actions/checkout@v4
@@ -382,7 +382,7 @@ jobs:
   publish_release:
     name: Publish Release
     needs: [get_version_code, publish_nuget, publish_npm, publish_admin]
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
     steps:
       - name: Create Github Release
         uses: softprops/action-gh-release@v1

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -154,7 +154,7 @@ jobs:
 
   windows_build_x86:
     name: Windows Build (x86)
-    runs-on: windows-2022
+    runs-on: windows-2025-vs2026
     steps:
       - name: Checkout Code
         uses: actions/checkout@v4
@@ -180,7 +180,7 @@ jobs:
 
   windows_build_x64:
     name: Windows Build (x64)
-    runs-on: windows-2022
+    runs-on: windows-2025-vs2026
     steps:
       - name: Checkout Code
         uses: actions/checkout@v4
@@ -317,7 +317,7 @@ jobs:
           #   version: "26.2"
           #   osFolder: "maccatalyst"
           - name: windows-os
-            os: windows-2022
+            os: windows-2025-vs2026
             version: "latest"
             osFolder: "windows"
 
@@ -385,7 +385,7 @@ jobs:
     runs-on: ubuntu-24.04
     steps:
       - name: Create Github Release
-        uses: softprops/action-gh-release@v1
+        uses: softprops/action-gh-release@v2
         with:
           name: Release ${{needs.get_version_code.outputs.version}}
           tag_name: ${{needs.get_version_code.outputs.version}}

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -313,8 +313,8 @@ jobs:
         name: ["windows-os"]
         include:
           # - name: macOS
-          #   os: macos-15
-          #   version: "16.4"
+          #   os: macos-26
+          #   version: "26.2"
           #   osFolder: "maccatalyst"
           - name: windows-os
             os: windows-2022

--- a/.github/workflows/sanitizer.yml
+++ b/.github/workflows/sanitizer.yml
@@ -20,7 +20,7 @@ env:
 
 jobs:
   Test:
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-24.04
 
     steps:
       - name: Fail if non Linux OS

--- a/Makefile
+++ b/Makefile
@@ -332,10 +332,14 @@ build-cli:
 	dotnet build -c $(TARGET) $(ELECTIONGUARD_APP_CLI_DIR)/ElectionGuard.CLI.sln /p:Platform=$(PROCESSOR)
 
 build-ui:
-# TODO: add platform checks, will not work on Linux!
 	@echo 🖥️ BUILD UI $(OPERATING_SYSTEM) $(PROCESSOR) $(TARGET)
 	cd ./src/electionguard-ui && dotnet restore
+ifeq ($(OPERATING_SYSTEM),Darwin)
+	dotnet build -f net10.0-maccatalyst -c $(TARGET) ./$(ELECTIONGUARD_APP_ADMIN_DIR)/ElectionGuard.UI/ElectionGuard.UI.csproj /p:APPCENTER_SECRET_MACOS=$(APPCENTER_SECRET_MACOS)
+endif
+ifeq ($(OPERATING_SYSTEM),Windows)
 	dotnet build -c $(TARGET) ./src/electionguard-ui/ElectionGuard.UI.sln /p:Platform=$(PROCESSOR) /p:APPCENTER_SECRET_UWP=$(APPCENTER_SECRET_UWP) /p:APPCENTER_SECRET_MACOS=$(APPCENTER_SECRET_MACOS)
+endif
 
 build-wasm:
 	@echo 🌐 BUILD WASM $(OPERATING_SYSTEM) $(PROCESSOR) $(TARGET)

--- a/Makefile
+++ b/Makefile
@@ -26,6 +26,7 @@ EMSDK?=$(ELECTIONGUARD_CACHE)/emscripten
 # not all platforms can compile all targets.
 # valid values:
 # OPERATING_SYSTEM: Android, Ios, Linux, Darwin, Windows
+# OS_VARIANT: Debian, Alpine
 # PROCESSOR: arm64, x64, x86
 ifeq ($(OS),Windows_NT)
 	OPERATING_SYSTEM ?= Windows
@@ -114,18 +115,42 @@ ifeq ($(OPERATING_SYSTEM),Darwin)
 endif
 ifeq ($(OPERATING_SYSTEM),Linux)
 	@echo 🐧 LINUX INSTALL
-	sudo apt install -y build-essential
-	sudo apt install -y iwyu
-	sudo apt install -y llvm
-	sudo apt install -y clang-12
-	sudo apt install -y cmake
-	sudo apt install -y lcov
-	sudo apt install -y cppcheck
-	sudo apt install -y clang-format
-	sudo apt install -y clang-tidy
-	sudo apt install -y ninja-build
-	sudo apt install -y valgrind
-	sudo apt install -y unzip
+	if [ -f /etc/os-release ] && grep -qi "alpine" /etc/os-release; then \
+		echo "🏔️ Alpine Linux detected"; \
+		sudo apk update; \		
+		echo "Installing CXX dependencies"; \
+		sudo apk add make musl-dev g++ cmake; \
+		echo "Installing DotNet dependencies"
+# See https://learn.microsoft.com/en-us/dotnet/core/install/linux-alpine?tabs=dotnet9
+		sudo apk add dotnet9-sdk dotnetcore9-runtime
+	else \
+		echo "🐧 Debian-based Linux detected"; \
+# to get Debian version of .NET
+		wget https://packages.microsoft.com/config/debian/12/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
+		sudo dpkg -i packages-microsoft-prod.deb
+		rm packages-microsoft-prod.deb
+		
+		sudo apt update; \
+		sudo apt install -y build-essential
+		sudo apt install -y iwyu
+		sudo apt install -y llvm
+# must update to bc bookworm doesn't have clang 12 anymore...
+# maybe don't specify a version unless we need older behavior??
+		sudo apt install -y clang-14
+		sudo apt install -y cmake
+		sudo apt install -y lcov
+		sudo apt install -y cppcheck
+		sudo apt install -y clang-format
+		sudo apt install -y clang-tidy
+		sudo apt install -y ninja-build
+		sudo apt install -y valgrind
+		sudo apt install -y unzip	
+# dotnet deps
+		sudo apt install -y dotnet-sdk-9.0
+# wasm deps
+		sudo apt install -y npm
+	fi
+	
 endif
 ifeq ($(OPERATING_SYSTEM),Windows)
 	@echo 🏁 WINDOWS INSTALL
@@ -134,10 +159,18 @@ ifeq ($(OPERATING_SYSTEM),Windows)
 	choco upgrade cmake -y
 	choco upgrade ninja -y
 	choco upgrade vswhere -y
+	choco upgrade llvm -y
 endif
 	wget -O cmake/CPM.cmake https://github.com/cpm-cmake/CPM.cmake/releases/download/v0.38.2/CPM.cmake
 	make fetch-sample-data
 	dotnet tool restore
+
+environment-dotnet:
+ifeq ($(OPERATING_SYSTEM),Linux)
+# Ubuntu 22.04 and later?
+	sudo add-apt-repository ppa:dotnet/backports
+	apt-get update &&   sudo apt-get install -y dotnet-sdk-7.0
+endif
 
 environment-msys2:
 ifeq ($(OPERATING_SYSTEM),Windows)
@@ -187,6 +220,7 @@ ifeq ($(OPERATING_SYSTEM),Windows)
 	cmake -S . -B $(ELECTIONGUARD_BUILD_LIBS_DIR)/$(OPERATING_SYSTEM)/$(PROCESSOR)/$(TARGET) \
 		-G "Visual Studio 18" -A $(VSPLATFORM) \
 		-DCMAKE_BUILD_TYPE=$(TARGET) \
+		-DCMAKE_OBJECT_PATH_MAX=200 \
 		-DBUILD_SHARED_LIBS=ON \
 		-DDISABLE_VALE=$(TEMP_DISABLE_VALE) \
 		-DUSE_MSVC=ON \
@@ -305,6 +339,7 @@ build-cli:
 	dotnet build -c $(TARGET) $(ELECTIONGUARD_APP_CLI_DIR)/ElectionGuard.CLI.sln /p:Platform=$(PROCESSOR)
 
 build-ui:
+# TODO: add platform checks, will not work on Linux!
 	@echo 🖥️ BUILD UI $(OPERATING_SYSTEM) $(PROCESSOR) $(TARGET)
 	cd ./src/electionguard-ui && dotnet restore
 	dotnet build -c $(TARGET) ./src/electionguard-ui/ElectionGuard.UI.sln /p:Platform=$(PROCESSOR) /p:APPCENTER_SECRET_UWP=$(APPCENTER_SECRET_UWP) /p:APPCENTER_SECRET_MACOS=$(APPCENTER_SECRET_MACOS)
@@ -388,7 +423,7 @@ generate-interop:
 # Lint / Format
 
 format: build
-	cd $(ELECTIONGUARD_BUILD_LIBS_DIR)/$(PROCESSOR) && $(MAKE) format
+	cd $(ELECTIONGUARD_BUILD_LIBS_DIR)/$(OPERATING_SYSTEM)/$(PROCESSOR) && $(MAKE) format
 
 lint:
 	dotnet jb inspectcode -o="lint-results.xml" -f="Xml" --build --verbosity="WARN" --severity="Warning" bindings/netstandard/ElectionGuard/ElectionGuard.sln
@@ -419,6 +454,7 @@ endif
 ifeq ($(OPERATING_SYSTEM),Darwin)
 	dotnet publish -f net10.0-maccatalyst -c $(TARGET) /p:CreatePackage=true /p:ApplicationVersion=$(BUILD_NUMBER) /p:APPCENTER_SECRET_MACOS=$(APPCENTER_SECRET_MACOS) ./$(ELECTIONGUARD_APP_ADMIN_DIR)/ElectionGuard.UI/ElectionGuard.UI.csproj -o ./publish
 endif
+# add error for running on Linuxx!
 
 publish-ui-appcenter: 
 	@echo 🧱 PUBLISH UI APPCENTER
@@ -440,7 +476,7 @@ else
 	@echo "APPCENTER_SECRET_MACOS not set. Skipping AppCenter publish"
 	exit 1
 endif
-
+# add linux check
 publish-wasm: build-npm
 	@echo 🌐 PUBLISH WASM
 ifeq ($(OPERATING_SYSTEM),Windows)
@@ -728,7 +764,7 @@ fetch-sample-data:
 	@echo ⬇️ FETCH Sample Data
 	wget -O $(TEMPFILE) https://github.com/microsoft/electionguard/releases/download/v1.0/sample-data.zip
 	unzip -o $(TEMPFILE)
-	rm -f $(TEMPFILE)
+#	rm -f $(TEMPFILE)
 
 generate-sample-data: build-netstandard
 	@echo Generate Sample Data

--- a/Makefile
+++ b/Makefile
@@ -117,40 +117,32 @@ ifeq ($(OPERATING_SYSTEM),Linux)
 	@echo 🐧 LINUX INSTALL
 	if [ -f /etc/os-release ] && grep -qi "alpine" /etc/os-release; then \
 		echo "🏔️ Alpine Linux detected"; \
-		sudo apk update; \		
+		sudo apk update; \
 		echo "Installing CXX dependencies"; \
 		sudo apk add make musl-dev g++ cmake; \
-		echo "Installing DotNet dependencies"
-# See https://learn.microsoft.com/en-us/dotnet/core/install/linux-alpine?tabs=dotnet9
-		sudo apk add dotnet9-sdk dotnetcore9-runtime
+		echo "Installing DotNet dependencies"; \
+		sudo apk add dotnet10-sdk dotnetcore10-runtime; \
 	else \
 		echo "🐧 Debian-based Linux detected"; \
-# to get Debian version of .NET
-		wget https://packages.microsoft.com/config/debian/12/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
-		sudo dpkg -i packages-microsoft-prod.deb
-		rm packages-microsoft-prod.deb
-		
+		wget https://packages.microsoft.com/config/debian/12/packages-microsoft-prod.deb -O packages-microsoft-prod.deb; \
+		sudo dpkg -i packages-microsoft-prod.deb; \
+		rm packages-microsoft-prod.deb; \
 		sudo apt update; \
-		sudo apt install -y build-essential
-		sudo apt install -y iwyu
-		sudo apt install -y llvm
-# must update to bc bookworm doesn't have clang 12 anymore...
-# maybe don't specify a version unless we need older behavior??
-		sudo apt install -y clang-14
-		sudo apt install -y cmake
-		sudo apt install -y lcov
-		sudo apt install -y cppcheck
-		sudo apt install -y clang-format
-		sudo apt install -y clang-tidy
-		sudo apt install -y ninja-build
-		sudo apt install -y valgrind
-		sudo apt install -y unzip	
-# dotnet deps
-		sudo apt install -y dotnet-sdk-9.0
-# wasm deps
-		sudo apt install -y npm
+		sudo apt install -y build-essential; \
+		sudo apt install -y iwyu; \
+		sudo apt install -y llvm; \
+		sudo apt install -y clang-14; \
+		sudo apt install -y cmake; \
+		sudo apt install -y lcov; \
+		sudo apt install -y cppcheck; \
+		sudo apt install -y clang-format; \
+		sudo apt install -y clang-tidy; \
+		sudo apt install -y ninja-build; \
+		sudo apt install -y valgrind; \
+		sudo apt install -y unzip; \
+		sudo apt install -y dotnet-sdk-10.0; \
+		sudo apt install -y npm; \
 	fi
-	
 endif
 ifeq ($(OPERATING_SYSTEM),Windows)
 	@echo 🏁 WINDOWS INSTALL
@@ -160,17 +152,18 @@ ifeq ($(OPERATING_SYSTEM),Windows)
 	choco upgrade ninja -y
 	choco upgrade vswhere -y
 	choco upgrade llvm -y
+	choco install dotnet-sdk
 endif
 	wget -O cmake/CPM.cmake https://github.com/cpm-cmake/CPM.cmake/releases/download/v0.38.2/CPM.cmake
 	make fetch-sample-data
 	dotnet tool restore
 
-environment-dotnet:
-ifeq ($(OPERATING_SYSTEM),Linux)
-# Ubuntu 22.04 and later?
-	sudo add-apt-repository ppa:dotnet/backports
-	apt-get update &&   sudo apt-get install -y dotnet-sdk-7.0
-endif
+# environment-dotnet:
+# ifeq ($(OPERATING_SYSTEM),Linux)
+# # Ubuntu 22.04 and later?
+# 	sudo add-apt-repository ppa:dotnet/backports
+# 	apt-get update &&   sudo apt-get install -y dotnet-sdk-10.0
+# endif
 
 environment-msys2:
 ifeq ($(OPERATING_SYSTEM),Windows)

--- a/cmake/tools.cmake
+++ b/cmake/tools.cmake
@@ -5,7 +5,7 @@
 option(CODE_COVERAGE "Enable code coverage" OFF)
 option(USE_STATIC_ANALYSIS "use static analysis tools" OFF)
 option(USE_DYNAMIC_ANALYSIS "use dynamic analysis tools" OFF)
-option(USE_FORMATTING "use formatting tools" OFF)
+option(USE_FORMATTING "use formatting tools" ON)
 
 function(use_valgrind TARGET_NAME)
     set(VALGRIND_LOG ${PROJECT_BINARY_DIR}/valgrind.log)
@@ -20,7 +20,7 @@ function(use_valgrind TARGET_NAME)
 endfunction()
 
 # ---- Dependencies ----
-include(cmake/CPM_0.31.0.cmake)
+include(cmake/CPM.cmake)
 
 CPMAddPackage(
     NAME StableCoder-cmake-scripts
@@ -111,7 +111,7 @@ if(USE_DYNAMIC_ANALYSIS)
     endif()
 endif()
 
-#if(USE_FORMATTING)
-#    message("++ Running with formatting")
-#    include(${StableCoder-cmake-scripts_SOURCE_DIR}/formatting.cmake)
-#endif()
+## if(USE_FORMATTING)
+##     message("++ Running with formatting")
+##     include(${StableCoder-cmake-scripts_SOURCE_DIR}/formatting.cmake)
+## endif()

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -4,7 +4,7 @@ cmake_minimum_required(VERSION 3.14...3.16 FATAL_ERROR)
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/../cmake")
 
 # ---- Dependencies ----
-include(../cmake/CPM_0.31.0.cmake)
+include(../cmake/CPM.cmake)
 
 CPMAddPackage(
     NAME date
@@ -266,7 +266,7 @@ if(CODE_COVERAGE)
     target_code_coverage(${META_PROJECT_TARGET} AUTO)
 endif()
 
-#if(USE_FORMATTING)
-#    message("++ Building with formatting")
-#    clang_format(format ${PROJECT_SOURCE_FILES})
-#endif()
+## if(USE_FORMATTING)
+##     message("++ Building with formatting")
+##     clang_format(format ${PROJECT_SOURCE_FILES})
+## endif()

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -14,9 +14,9 @@ CPMAddPackage(
 
 CPMAddPackage(
     NAME nlohmann_json
-    VERSION 3.11.2
-    URL https://github.com/nlohmann/json/releases/download/v3.11.2/include.zip
-    URL_HASH SHA256=e5c7a9f49a16814be27e4ed0ee900ecd0092bfb7dbfca65b5a421b774dccaaed
+    VERSION 3.11.3
+    URL https://github.com/nlohmann/json/releases/download/v3.11.3/include.zip
+    URL_HASH SHA256=a22461d13119ac5c78f205d3df1db13403e58ce1bb1794edc9313677313f4a9d
 )
 
 CPMAddPackage(

--- a/src/electionguard-ui/Directory.Packages.props
+++ b/src/electionguard-ui/Directory.Packages.props
@@ -8,8 +8,9 @@
     <PackageVersion Include="Microsoft.AppCenter.Crashes" Version="5.0.2" />
     <PackageVersion Include="Microsoft.AppCenter.Distribute" Version="5.0.2" />
     <PackageVersion Include="Microsoft.Extensions.Logging.Debug" Version="7.0.0" />
-    <PackageVersion Include="Microsoft.Maui.Controls" Version="9.0.40" />
-    <PackageVersion Include="Microsoft.Maui.Controls.Xaml" Version="9.0.40" />
+    <PackageVersion Include="Microsoft.Maui.Controls" Version="8.0.93" />
+    <PackageVersion Include="Microsoft.Maui.Core" Version="9.0.21" />
+    <PackageVersion Include="Microsoft.UI.Xaml" Version="2.8.6" />
     <PackageVersion Include="MongoDB.Driver" Version="2.19.1" />
     <PackageVersion Include="Newtonsoft.Json" Version="13.0.3" />
     <PackageVersion Include="Shouldly" Version="4.1.0" />

--- a/src/electionguard-ui/ElectionGuard.UI.Lib/Models/ElectionPartialKeyBackup.cs
+++ b/src/electionguard-ui/ElectionGuard.UI.Lib/Models/ElectionPartialKeyBackup.cs
@@ -54,7 +54,7 @@ public class ElectionPartialKeyBackup : DisposableBase
         OwnerSequenceOrder = contract.OwnerSequenceOrder;
         DesignatedId = contract.DesignatedId;
         DesignatedSequenceOrder = contract.DesignatedSequenceOrder;
-        EncryptedCoordinate = contract.EncryptedCoordinate;
+        EncryptedCoordinate = new(contract.EncryptedCoordinate);
     }
 
     public ElectionPartialKeyBackupRecord ToRecord()

--- a/src/electionguard-ui/ElectionGuard.UI/ElectionGuard.UI.csproj
+++ b/src/electionguard-ui/ElectionGuard.UI/ElectionGuard.UI.csproj
@@ -69,7 +69,7 @@
 	<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
 		<IntermediateOutputPath></IntermediateOutputPath>
 		<DebugSymbols></DebugSymbols>
-		<DebugType></DebugType>
+		<DebugType>portable</DebugType>
 		<Optimize></Optimize>
 		<OutputPath></OutputPath>
 		<DefineConstants></DefineConstants>
@@ -96,7 +96,7 @@
 	<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
 		<IntermediateOutputPath></IntermediateOutputPath>
 		<DebugSymbols></DebugSymbols>
-		<DebugType></DebugType>
+		<DebugType>portable</DebugType>
 		<Optimize></Optimize>
 		<OutputPath></OutputPath>
 		<DefineConstants></DefineConstants>

--- a/src/electionguard-ui/ElectionGuard.UI/ElectionGuard.UI.csproj
+++ b/src/electionguard-ui/ElectionGuard.UI/ElectionGuard.UI.csproj
@@ -1,8 +1,8 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
 	<PropertyGroup>
-		<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('OSX'))">net7.0-maccatalyst</TargetFrameworks>
-		<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">net7.0-windows10.0.19041.0</TargetFrameworks>
+		<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('OSX'))">net10.0-maccatalyst</TargetFrameworks>
+		<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">net10.0-windows10.0.19041.0</TargetFrameworks>
 		<OutputType>Exe</OutputType>
 		<ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
 		<UseMaui>true</UseMaui>
@@ -38,31 +38,31 @@
 	</PropertyGroup>
 
 
-	<PropertyGroup Label="Debug|arm64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net7.0-maccatalyst|arm64'">
+	<PropertyGroup Label="Debug|arm64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net10.0-maccatalyst|arm64'">
 		<CreatePackage>false</CreatePackage>
 		<RuntimeIdentifiers>maccatalyst-arm64;maccatalyst-x64</RuntimeIdentifiers>
 	</PropertyGroup>
 
-	<PropertyGroup Label="Release|arm64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net7.0-maccatalyst|arm64'">
+	<PropertyGroup Label="Release|arm64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net10.0-maccatalyst|arm64'">
 		<CreatePackage>true</CreatePackage>
 		<RuntimeIdentifiers>maccatalyst-arm64;maccatalyst-x64</RuntimeIdentifiers>
 	</PropertyGroup>
 
-	<PropertyGroup Label="Debug|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net7.0-maccatalyst|x64'">
+	<PropertyGroup Label="Debug|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net10.0-maccatalyst|x64'">
 		<CreatePackage>false</CreatePackage>
 		<RuntimeIdentifiers>maccatalyst-x64</RuntimeIdentifiers>
 	</PropertyGroup>
 
-	<PropertyGroup Label="Release|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net7.0-maccatalyst|x64'">
+	<PropertyGroup Label="Release|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net10.0-maccatalyst|x64'">
 		<CreatePackage>false</CreatePackage>
 		<RuntimeIdentifiers>maccatalyst-x64</RuntimeIdentifiers>
 	</PropertyGroup>
 
-	<PropertyGroup Label="Debug|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net7.0-windows10.0.19041.0|x64'">
+	<PropertyGroup Label="Debug|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net10.0-windows10.0.19041.0|x64'">
 		<CreatePackage>false</CreatePackage>
 	</PropertyGroup>
 
-	<PropertyGroup Label="Release|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net7.0-windows10.0.19041.0|x64'">
+	<PropertyGroup Label="Release|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net10.0-windows10.0.19041.0|x64'">
 		<CreatePackage>true</CreatePackage>
 	</PropertyGroup>
 
@@ -222,11 +222,11 @@
 		<ElectionGuardLibs>..\..\..\build\libs</ElectionGuardLibs>
 		<ElectionGuardScripts>..\..\..\scripts</ElectionGuardScripts>
 	</PropertyGroup>
-	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net7.0-windows10.0.19041.0|AnyCPU'">
+	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net10.0-windows10.0.19041.0|AnyCPU'">
 		<DefineConstants>$(DefineConstants);DISABLE_XAML_GENERATED_BREAK_ON_UNHANDLED_EXCEPTION</DefineConstants>
 		<DebugType>portable</DebugType>
 	</PropertyGroup>
-	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net7.0-windows10.0.19041.0|AnyCPU'">
+	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net10.0-windows10.0.19041.0|AnyCPU'">
 		<DefineConstants>$(DefineConstants);DISABLE_XAML_GENERATED_BREAK_ON_UNHANDLED_EXCEPTION</DefineConstants>
 		<DebugType>portable</DebugType>
 	</PropertyGroup>

--- a/src/electionguard-ui/ElectionGuard.UI/ElectionGuard.UI.csproj
+++ b/src/electionguard-ui/ElectionGuard.UI/ElectionGuard.UI.csproj
@@ -23,6 +23,8 @@
 
 		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'">15.3</SupportedOSPlatformVersion>
 		<RuntimeIdentifiers Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'">maccatalyst-arm64</RuntimeIdentifiers>
+		<!-- Workaround: ILLink crashes with NullReferenceException on .NET 10 SDK (ShouldTreatWarningAsError bug) -->
+		<MtouchLink Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'">None</MtouchLink>
 		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.19041.0</SupportedOSPlatformVersion>
 		<TargetPlatformMinVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.19041.0</TargetPlatformMinVersion>
 		<Nullable>enable</Nullable>

--- a/src/electionguard-ui/ElectionGuard.UI/ElectionGuard.UI.csproj
+++ b/src/electionguard-ui/ElectionGuard.UI/ElectionGuard.UI.csproj
@@ -40,12 +40,12 @@
 
 	<PropertyGroup Label="Debug|arm64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net10.0-maccatalyst|arm64'">
 		<CreatePackage>false</CreatePackage>
-		<RuntimeIdentifiers>maccatalyst-arm64;maccatalyst-x64</RuntimeIdentifiers>
+		<RuntimeIdentifiers>maccatalyst-arm64</RuntimeIdentifiers>
 	</PropertyGroup>
 
 	<PropertyGroup Label="Release|arm64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net10.0-maccatalyst|arm64'">
 		<CreatePackage>true</CreatePackage>
-		<RuntimeIdentifiers>maccatalyst-arm64;maccatalyst-x64</RuntimeIdentifiers>
+		<RuntimeIdentifiers>maccatalyst-arm64</RuntimeIdentifiers>
 	</PropertyGroup>
 
 	<PropertyGroup Label="Debug|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net10.0-maccatalyst|x64'">

--- a/src/electionguard-ui/ElectionGuard.UI/ElectionGuard.UI.csproj
+++ b/src/electionguard-ui/ElectionGuard.UI/ElectionGuard.UI.csproj
@@ -22,6 +22,7 @@
 		<ApplicationVersion>19</ApplicationVersion>
 
 		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'">15.3</SupportedOSPlatformVersion>
+		<RuntimeIdentifiers Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'">maccatalyst-arm64</RuntimeIdentifiers>
 		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.19041.0</SupportedOSPlatformVersion>
 		<TargetPlatformMinVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.19041.0</TargetPlatformMinVersion>
 		<Nullable>enable</Nullable>

--- a/src/electionguard-ui/ElectionGuard.UI/ElectionGuard.UI.csproj
+++ b/src/electionguard-ui/ElectionGuard.UI/ElectionGuard.UI.csproj
@@ -28,7 +28,7 @@
 		<CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
 		<RestoreProjectStyle>PackageReference</RestoreProjectStyle>
 		<!--<Configurations>Release;Debug</Configurations>-->
-		<Platforms Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'">arm64;x64</Platforms>
+		<Platforms Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'">arm64</Platforms>
 		<Platforms Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">AnyCPU</Platforms>
 		<UserSecretsId>efb177b9-29c3-4df1-adaa-be7e64bdc8ce</UserSecretsId>
 	</PropertyGroup>
@@ -46,16 +46,6 @@
 	<PropertyGroup Label="Release|arm64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net10.0-maccatalyst|arm64'">
 		<CreatePackage>true</CreatePackage>
 		<RuntimeIdentifiers>maccatalyst-arm64</RuntimeIdentifiers>
-	</PropertyGroup>
-
-	<PropertyGroup Label="Debug|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net10.0-maccatalyst|x64'">
-		<CreatePackage>false</CreatePackage>
-		<RuntimeIdentifiers>maccatalyst-x64</RuntimeIdentifiers>
-	</PropertyGroup>
-
-	<PropertyGroup Label="Release|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net10.0-maccatalyst|x64'">
-		<CreatePackage>false</CreatePackage>
-		<RuntimeIdentifiers>maccatalyst-x64</RuntimeIdentifiers>
 	</PropertyGroup>
 
 	<PropertyGroup Label="Debug|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net10.0-windows10.0.19041.0|x64'">

--- a/src/electionguard-ui/ElectionGuard.UI/ElectionGuard.UI.csproj
+++ b/src/electionguard-ui/ElectionGuard.UI/ElectionGuard.UI.csproj
@@ -1,8 +1,8 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
 	<PropertyGroup>
-		
-		<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">net10.0-windows10.0.19041.0</TargetFrameworks>
+		<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('OSX'))">net7.0-maccatalyst</TargetFrameworks>
+		<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">net7.0-windows10.0.19041.0</TargetFrameworks>
 		<OutputType>Exe</OutputType>
 		<ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
 		<UseMaui>true</UseMaui>
@@ -20,7 +20,7 @@
 		<!-- Versions -->
 		<ApplicationDisplayVersion>1.92.0</ApplicationDisplayVersion>
 		<ApplicationVersion>19</ApplicationVersion>
-		
+
 		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'">15.3</SupportedOSPlatformVersion>
 		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.19041.0</SupportedOSPlatformVersion>
 		<TargetPlatformMinVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.19041.0</TargetPlatformMinVersion>
@@ -37,31 +37,32 @@
 		<DefineConstants>$(DefineConstants);APPCENTER_SECRET_MACOS=$(APPCENTER_SECRET_MACOS);APPCENTER_SECRET_UWP=$(APPCENTER_SECRET_UWP)</DefineConstants>
 	</PropertyGroup>
 
-	<PropertyGroup Label="Debug|arm64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net10.0-maccatalyst|arm64'">
+
+	<PropertyGroup Label="Debug|arm64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net7.0-maccatalyst|arm64'">
 		<CreatePackage>false</CreatePackage>
 		<RuntimeIdentifiers>maccatalyst-arm64;maccatalyst-x64</RuntimeIdentifiers>
 	</PropertyGroup>
 
-	<PropertyGroup Label="Release|arm64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net10.0-maccatalyst|arm64'">
+	<PropertyGroup Label="Release|arm64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net7.0-maccatalyst|arm64'">
 		<CreatePackage>true</CreatePackage>
 		<RuntimeIdentifiers>maccatalyst-arm64;maccatalyst-x64</RuntimeIdentifiers>
 	</PropertyGroup>
 
-	<PropertyGroup Label="Debug|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net10.0-maccatalyst|x64'">
+	<PropertyGroup Label="Debug|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net7.0-maccatalyst|x64'">
 		<CreatePackage>false</CreatePackage>
 		<RuntimeIdentifiers>maccatalyst-x64</RuntimeIdentifiers>
 	</PropertyGroup>
 
-	<PropertyGroup Label="Release|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net10.0-maccatalyst|x64'">
+	<PropertyGroup Label="Release|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net7.0-maccatalyst|x64'">
 		<CreatePackage>false</CreatePackage>
 		<RuntimeIdentifiers>maccatalyst-x64</RuntimeIdentifiers>
 	</PropertyGroup>
 
-	<PropertyGroup Label="Debug|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net10.0-windows10.0.19041.0|x64'">
+	<PropertyGroup Label="Debug|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net7.0-windows10.0.19041.0|x64'">
 		<CreatePackage>false</CreatePackage>
 	</PropertyGroup>
 
-	<PropertyGroup Label="Release|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net10.0-windows10.0.19041.0|x64'">
+	<PropertyGroup Label="Release|x64" Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net7.0-windows10.0.19041.0|x64'">
 		<CreatePackage>true</CreatePackage>
 	</PropertyGroup>
 
@@ -69,7 +70,7 @@
 		<IntermediateOutputPath></IntermediateOutputPath>
 		<DebugSymbols></DebugSymbols>
 		<DebugType></DebugType>
-		<Optimize>False</Optimize>
+		<Optimize></Optimize>
 		<OutputPath></OutputPath>
 		<DefineConstants></DefineConstants>
 		<WarningLevel></WarningLevel>
@@ -96,7 +97,7 @@
 		<IntermediateOutputPath></IntermediateOutputPath>
 		<DebugSymbols></DebugSymbols>
 		<DebugType></DebugType>
-		<Optimize>False</Optimize>
+		<Optimize></Optimize>
 		<OutputPath></OutputPath>
 		<DefineConstants></DefineConstants>
 		<WarningLevel></WarningLevel>
@@ -149,11 +150,12 @@
 		<PackageReference Include="CommunityToolkit.Maui" />
 		<PackageReference Include="CommunityToolkit.Mvvm" />
 		<PackageReference Include="Microsoft.Maui.Controls" />
-		<PackageReference Include="Microsoft.Maui.Controls.Xaml" />
+		<PackageReference Include="Microsoft.Maui.Core" />
 	</ItemGroup>
 
 	<ItemGroup>
 		<ProjectReference Include="..\..\..\bindings\netstandard\ElectionGuard\ElectionGuard.Decryption\ElectionGuard.Decryption.csproj" />
+		<ProjectReference Include="..\..\..\bindings\netstandard\ElectionGuard\ElectionGuard.ElectionSetup\ElectionGuard.ElectionSetup.csproj" />
 		<ProjectReference Include="..\..\..\bindings\netstandard\ElectionGuard\ElectionGuard.Encryption\ElectionGuard.Encryption.csproj">
 			<!-- <CopyLocalSatelliteAssemblies>True</CopyLocalSatelliteAssemblies>
 	    <Private>True</Private> -->
@@ -220,11 +222,11 @@
 		<ElectionGuardLibs>..\..\..\build\libs</ElectionGuardLibs>
 		<ElectionGuardScripts>..\..\..\scripts</ElectionGuardScripts>
 	</PropertyGroup>
-	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net10.0-windows10.0.19041.0|AnyCPU'">
+	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net7.0-windows10.0.19041.0|AnyCPU'">
 		<DefineConstants>$(DefineConstants);DISABLE_XAML_GENERATED_BREAK_ON_UNHANDLED_EXCEPTION</DefineConstants>
 		<DebugType>portable</DebugType>
 	</PropertyGroup>
-	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net10.0-windows10.0.19041.0|AnyCPU'">
+	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net7.0-windows10.0.19041.0|AnyCPU'">
 		<DefineConstants>$(DefineConstants);DISABLE_XAML_GENERATED_BREAK_ON_UNHANDLED_EXCEPTION</DefineConstants>
 		<DebugType>portable</DebugType>
 	</PropertyGroup>

--- a/src/electionguard-ui/ElectionGuard.UI/Services/TallyManager.cs
+++ b/src/electionguard-ui/ElectionGuard.UI/Services/TallyManager.cs
@@ -5,12 +5,12 @@ using ElectionGuard.Decryption.Decryption;
 using ElectionGuard.Decryption.Extensions;
 using ElectionGuard.Decryption.Shares;
 using ElectionGuard.Decryption.Tally;
+using ElectionGuard.ElectionSetup;
 using ElectionGuard.Converters;
 using ElectionGuard.Extensions;
 using ElectionGuard.Guardians;
 using MongoDB.Driver;
 using Newtonsoft.Json;
-using ElectionGuard.ElectionSetup;
 
 namespace ElectionGuard.UI.Services
 {
@@ -381,7 +381,7 @@ namespace ElectionGuard.UI.Services
                     userId,
                     keyCeremony,
                     publicKeys,
-                    backups.ToDictionary(k => k.GuardianId!, v => v.Backup!.ToRecord())) ?? throw new ElectionGuardException(nameof(userId));
+                    backups.ToDictionary(k => k.GuardianId!, v => v.Backup!)) ?? throw new ElectionGuardException(nameof(userId));
                 return guardian;
             }
             catch (Exception ex)

--- a/src/electionguard-ui/ElectionGuard.UI/Services/TallyManager.cs
+++ b/src/electionguard-ui/ElectionGuard.UI/Services/TallyManager.cs
@@ -381,7 +381,7 @@ namespace ElectionGuard.UI.Services
                     userId,
                     keyCeremony,
                     publicKeys,
-                    backups.ToDictionary(k => k.GuardianId!, v => v.Backup!)) ?? throw new ElectionGuardException(nameof(userId));
+                    backups.ToDictionary(k => k.GuardianId!, v => v.Backup!.ToRecord())) ?? throw new ElectionGuardException(nameof(userId));
                 return guardian;
             }
             catch (Exception ex)

--- a/src/electionguard/lookup_table.hpp
+++ b/src/electionguard/lookup_table.hpp
@@ -5,6 +5,7 @@
 #include "facades/bignum4096.hpp"
 #include "utils.hpp"
 
+#include <algorithm>
 #include <array>
 #include <cstdint>
 #include <electionguard/async.hpp>

